pkg/catalog/catalog.go

```diff
@@ -199,12 +205,13 @@ func New(ctx context.Context, cfg Config) (*Catalog, error) {
 	protectedBranchesManager := branch.NewProtectionManager(settingManager)
 	stagingManager := staging.NewManager(ctx, *cfg.KVStore)
 	gStore := graveler.NewKVGraveler(committedManager, stagingManager, refManager, gcManager, protectedBranchesManager)
-
+	workpool := pond.New(sharedWorkers, sharedWorkers*workersToBufferRatio)
```

Comment:
Might want to initialize the pool with the parent context
```suggestion
	workpool := pond.New(sharedWorkers, sharedWorkers*workersToBufferRatio, pond.Context(ctx))
```
Also, did you try to play around with the resize strategy and see which provides the best performance?

Reply:
What about the resize strategy? Have you looked into it?

Reply:
Sorry, missed that. What do you mean by resize strategy? `pond` can scale up to the given maximum number of workers. It's not just a matter of performance, we should also consider starvation for other requests too. I assumed that capping it to 15 would allow serving several (up to 15) LogCommits concurrent requests, while not hurting performance of other requests. What measurements would you like me to perform?

Reply:
See documentation at: https://github.com/alitto/pond

> 
> - Strategy: Configures the strategy used to resize the pool when backpressure is detected. You can create a custom strategy by implementing the pond.ResizingStrategy interface or choose one of the 3 presets:
>   - Eager: maximizes responsiveness at the expense of higher resource usage, which can reduce throughput under certain conditions. This strategy is meant for worker pools that will operate at a small percentage of their capacity most of the time and may occasionally receive bursts of tasks. This is the default strategy.
>   - Balanced: tries to find a balance between responsiveness and throughput. It's suitable for general purpose worker pools or those that will operate close to 50% of their capacity most of the time.
>   - Lazy: maximizes throughput at the expense of responsiveness. This strategy is meant for worker pools that will operate close to their max. capacity most of the time.

Reply:
Thanks for the explanation - I did miss that.
The default strategy is `Eager` and I think that it suits LogCommits usages best of all 3.